modules/optimization/include/dirackernels/ReporterTimePointSource.h

```diff
@@ -10,37 +10,23 @@
 #pragma once
 
 // Moose Includes
-#include "DiracKernel.h"
-#include "ReporterInterface.h"
+#include "ReporterPointSource.h"
 
 /**
- * Apply a point load defined by vectors.
+ * Apply a time dependent point load defined by Reporters.
  */
-class ReporterTimePointSource : public DiracKernel, public ReporterInterface
+class ReporterTimePointSource : public ReporterPointSource
 {
 public:
   static InputParameters validParams();
   ReporterTimePointSource(const InputParameters & parameters);
   virtual void addPoints() override;
-
 protected:
-  virtual Real computeQpResidual() override;
-
-private:
-  /// x-coordinates from reporter
-  const std::vector<Real> & _coordx;
-  /// y-coordinates from reporter
-  const std::vector<Real> & _coordy;
-  /// z-coordinates from reporter
-  const std::vector<Real> & _coordz;
   /// time-coordinates from reporter
   const std::vector<Real> & _coordt;
-  /// values from reporter
-  const std::vector<Real> & _values;
+
+private:
```

Comment:
Why did you add this?

Reply:
I didn't add the private, you already had it private.